jaeger-core/src/main/java/com/uber/jaeger/samplers/GuaranteedThroughputSampler.java

```diff
@@ -37,11 +43,39 @@
 @EqualsAndHashCode
 @AllArgsConstructor (access = AccessLevel.PACKAGE) // Visible for testing
 public final class GuaranteedThroughputSampler implements Sampler {
-  private final ProbabilisticSampler probabilisticSampler;
-  private final RateLimitingSampler lowerBoundSampler;
+  public static final String TYPE = "lowerbound";
+
+  private ProbabilisticSampler probabilisticSampler;
```

Comment:
I prefer immutable objects to mutable ones, and would prefer that these fields are final. 
What problem are we trying to solve by having an `update` method?